src/custom/utils/trade.ts

```diff
@@ -24,16 +24,49 @@ export interface PostOrderParams {
   recipientAddressOrName: string | null
   addPendingOrder: (order: AddPendingOrderParams) => void
 }
+interface DetermineInputOutputParams<T> {
+  kind: string | 'sell' | 'buy'
+  inputAmount: T
+  outputAmount: T
+}
+
+export function determineInputOutput({
+  kind,
+  inputAmount,
+  outputAmount
+}: DetermineInputOutputParams<string>): { input: string; output: string; quantifier: string }
+export function determineInputOutput({
+  kind,
+  inputAmount,
+  outputAmount
+}: DetermineInputOutputParams<CurrencyAmount>): { input: CurrencyAmount; output: CurrencyAmount; quantifier: string }
+export function determineInputOutput({
+  kind,
+  inputAmount,
+  outputAmount
+}: DetermineInputOutputParams<CurrencyAmount | string>) {
+  const input = kind === 'sell' ? inputAmount : outputAmount
+  const output = kind === 'sell' ? outputAmount : inputAmount
+  const quantifier = kind === 'sell' ? 'at least' : 'at most'
```

Comment:
Nitpick:

What about
```
if (kind === 'sell') {
  input = inputAmount
  output = outputAmount
  quantifier = 'at least'
} else {
 // you get the point
}
```
Instead of 3 inline identical ifs?
More lines, yes, but grouped.

Not a big deal though.

Reply:
yeah i had this at first but i guess i preferred the shorter approach. if it's an issue i can change it but I think I will leave it as is